pkg/ovsdb/monitor.go

```diff
@@ -480,3 +527,45 @@ func (u *updater) prepareRow(value []byte) (map[string]interface{}, string, erro
 	u.deleteUnselectedColumns(data)
 	return data, uuid, nil
 }
+
+func columnArrayToMap(columns []string, columnSchemas map[string]*libovsdb.ColumnSchema) map[string]*libovsdb.ColumnSchema {
+	ret := map[string]*libovsdb.ColumnSchema{}
+	if len(columns) != 0 {
+		for _, column := range columns {
+			ret[column] = columnSchemas[column]
+		}
+	} else {
+		for column, schema := range columnSchemas {
+			ret[column] = schema
+		}
+	}
+	return ret
+}
+
+func difference(set1 libovsdb.OvsSet, set2 libovsdb.OvsSet) libovsdb.OvsSet {
+	var diff libovsdb.OvsSet
+
+	// Loop two times, first to find slice1 strings not in slice2,
+	// second loop to find slice2 strings not in slice1
+	for i := 0; i < 2; i++ {
+		for _, s1 := range set1.GoSet {
+			found := false
+			for _, s2 := range set2.GoSet {
+				if s1 == s2 {
```

Comment:
is there relationship between s1 and s2 i.e one of them is subset of the other ? because here u check for the 1st matching element and stop there could be another mismatching fields after ?

Reply:
the sets do not have any relationship, and I check them 2, see the swap operation at the end of the function. The break doesn't stop on the first matching element, it just breaks the internal loop.
See the tests that I just added to the monitor_test.go.
 
P.S. the PR is not ready